llvm/lib/Target/AMDGPU/GCNRegPressure.cpp

```diff
@@ -274,32 +274,42 @@ void GCNUpwardRPTracker::recede(const MachineInstr &MI) {
   if (MI.isDebugInstr())
     return;
 
-  auto DecrementDef = [this](const MachineOperand &MO) {
+  // Kill all defs.
+  GCNRegPressure DefPressure, ECDefPressure;
+  bool HasECDefs = false;
```

Comment:
Just to check my understanding - adding the variable `HasECDefs` is not really needed, but it saves some cpu cycles as it avoids adding pressures with all zeros in a common non-clobber case (`DefPressure += ECDefPressure`). Is that right?

Reply:
Right, I decided to add it because early-clobbers are rare.

Reply:
Maybe not worth it, since `GCNRegPressure::operator+` is just 6 integer add instructions - but I don't really mind either way.